src/app/business/profile/profile.component.ts

```diff
@@ -68,12 +72,51 @@ export class ProfileComponent implements OnInit {
     }
     getProfiles() {
         this.ProfileService.getProfiles().subscribe((res) => {
-            this.profiles = res.json();
-            this.profiles.forEach(item=>{
+            let str = res.json();
+            this.allProfileNameForCheck = [];
+            str.forEach(item=>{
                 if(item.storageType && item.storageType == "file"){
                     delete item.replicationProperties;
                 }
+                this.allProfileNameForCheck.push(item.name);
             })
+            this.getPools(str);
         });
     }
+    getPools(str) {
+        this.PoolService.getPools().subscribe((res) => {
+            this.pools = res.json();
+            str.forEach(item => {
+                item.totalFreeCapacity = this.getSumCapacity(this.pools, 'free',item);
+                item.totalCapacity = this.getSumCapacity(this.pools, 'total',item);
+            });
+            this.profiles = str;
+        },
+        err=>{
+            this.profiles = str;
+        });
+    }
+    getSumCapacity(pools, FreeOrTotal,data) {
+        let SumCapacity: number = 0;
+        let newPools = lodash.cloneDeep(pools);
+        newPools = newPools.filter(item=>{
+            return item.storageType == data.storageType;
+        })
+        let arrLength = newPools.length;
+        for (let i = 0; i < arrLength; i++) {
+            let valid = data && data["provisioningProperties"].ioConnectivity.accessProtocol && data["provisioningProperties"].ioConnectivity.accessProtocol.toLowerCase() 
+            == newPools[i].extras.ioConnectivity.accessProtocol;
+            let blockValid = data["provisioningProperties"].dataStorage.provisioningPolicy == newPools[i].extras.dataStorage.provisioningPolicy;
+            if(valid && (data.storageType == "file" ||  (data.storageType == "block" && blockValid))){
```

Comment:
since we are looping the list of pools anyways, better calculate sumFree and sumTotal and keep in some local variables, use it each time, instead of calculating each time

Reply:
I just something in the new API spec that we are discussing and some of these fields will be part of the API response. Consider LGTM.